### ISSUE TITLE
Creating products with generator fails

### ISSUE THREAD

**Issue by twojtylak** [solutionDrive/SyliusProductBundlesPlugin#24]
In the product entity the field `code` is mandatory, so calling:

````
$productBundle = $this->productBundleCreator->createProductBundle($product->getName(), $product);
$em->persist();
$em->flush();
````
fails. 

`[PDOException (23000)]                                                                                             
  SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: solutiondrive_product_bundle.code  
`

So either we 
- remove the code field as suggested in #23 
- or the code should be passed as argument
- or the code should be taken from Product in  createProductBundle

https://github.com/solutionDrive/SyliusProductBundlesPlugin/blob/84d57030be3646700c18a98d4c3d1df413625c48/src/Service/ProductBundleCreator.php#L37-L44

**Comment by mattagohni** [comment]
I just had a look where the CodeAwareInterface is used and it seems on first glance almost every Resource-Interface extends the CodeAwareInterface. So maybe there is a good chance that it is not the best idea to remove it entierely.
![bildschirmfoto 2018-03-13 um 10 23 30](https://user-images.githubusercontent.com/12425528/37333186-f7be825e-26a8-11e8-949f-481bbd0b360d.png)

**Comment by mattagohni** [comment]
If we would take the name and the code from the product (maybe with a fixed pre- or suffix to have it unique throughout the system), we could also get rid of the first parameter for the createProductBundle-Method.

**Comment by mattagohni** [comment]
@twojtylak is this also fixed by #27 ?

**Comment by twojtylak** [comment]
yes
